### DISCUSSION TITLE
2021-11-08 Core Team Update

### DISCUSSION THREAD

**Discussion by mm-near** [near/community#122] [Core]
# Highlights:
* sharding seems to be stable in testnet, preparing for mainnet launch

# TL;DR
@pmnoxx - improving network performance
@Longarithm - improving fees
@mzhangmzz - improvements/fixes before sharding mainnet launch
@yordanmadzhunkov - improve fast-testing (have deterministic tests) before our 'more validators' launch on Nov 29
@EgorKulikov - working on different validator assignment (for 'more validators' on Nov 29)
@abacabadabacaba - better storage analysis (replace Merkle tree with AVL)

# Done:
@pmnoxx - larger refactoring of the networking code (to move routing table to separate thread and improve performance): https://github.com/near/nearcore/pull/5137, https://github.com/near/nearcore/pull/5141, https://github.com/near/nearcore/pull/5140, https://github.com/near/nearcore/pull/5089

@Longarithm - fixed regular_op_cost fee, looking into old vs new fees discrepancy (https://github.com/near/nearcore/pull/4733#issuecomment-954919299) 
@mzhangmzz - fixing nayduck failures (https://github.com/near/nearcore/pull/5154), 
@mzhangmzz - fixing issues in view client (blocking actor - https://github.com/near/nearcore/pull/5125), (GetExecutionOption - https://github.com/near/nearcore/pull/5133)

# Progress:
@pmnoxx - reference level documentation for network - https://github.com/near/nearcore/pull/5156
@pmnoxx - tracking memory usage in peers https://github.com/near/nearcore/pull/5155
@pmnoxx - design of throttling during reading: https://github.com/near/nearcore/pull/4927

@Longarithm - preparing summary of the possible solutions for function-related fees
@Longarithm - working on the genesis/runtime working separation(https://github.com/near/nearcore/pull/4447)

@mzhangmzz - investigating the issue with rpc nodes getting out of sync (https://github.com/near/nearcore/pull/5112)
@yordanmadzhunkov - designing a mock for the clock call (https://github.com/near/nearcore/pull/5123) and converting tests to be deterministic
@abacabadabacaba - working on the AVL tree implementation

**Comment by maxzaver** [comment]
@mm-near 

> @Longarithm - fixed regular_op_cost fee, looking into old vs new fees discrepancy (near/nearcore#4733 (comment))

Which PR or commit is this referring to?

> @abacabadabacaba - better storage analysis (replace Merkle tree with AVL)

Are there intermediate results to look at? Also what was achieved compared with 1 week ago?

**Reply by Longarithm** [reply]
@nearmax regular_op_cost fix refers to https://github.com/near/nearcore/pull/5099 and fixes https://github.com/near/nearcore/issues/5094.
